### PR TITLE
use yaml mime.

### PR DESCRIPTION
**Pull request by vwxyzh**
@chenkennt @superyyrrzz @hellosnow @ansyral @qinezh

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.ManagedReference/ManagedReferenceDocumentProcessor.cs [hunk 1]
```diff
@@ -31,16 +31,27 @@ public override ProcessingPriority GetProcessingPriority(FileAndType file)
             switch (file.Type)
             {
                 case DocumentType.Article:
-                    if (".yml".Equals(Path.GetExtension(file.File), StringComparison.OrdinalIgnoreCase))
+                    if (".yml".Equals(Path.GetExtension(file.File), StringComparison.OrdinalIgnoreCase) ||
+                        ".yaml".Equals(Path.GetExtension(file.File), StringComparison.OrdinalIgnoreCase))
                     {
-                        return ProcessingPriority.Normal;
+                        var mime = YamlMime.ReadMime(Path.Combine(file.BaseDir, file.File));
```
Comment (chenkennt) [inline src/Microsoft.DocAsCode.Build.ManagedReference/ManagedReferenceDocumentProcessor.cs]: >ReadMime [](start = 44, length = 8)

I looked at ReadMime code, seems to be too simple. What if the first line is a normal yaml comment? Should a more strict format for yaml mime type (for example, must start with #yamlmime)?

Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Build.ManagedReference/ManagedReferenceDocumentProcessor.cs]: now the mime must start with "YamlMime:"

---
In reply to: [73289646](https://github.com/dotnet/docfx/pull/562#discussion_r73289646) [](ancestors = 73289646)

**Comment by chenkennt** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Common/YamlMime.cs [hunk 2]
```diff
@@ -8,9 +8,10 @@ namespace Microsoft.DocAsCode.Common
 
     public static class YamlMime
     {
-        public const string ManagedReference = nameof(YamlMime) + ":" + nameof(ManagedReference);
-        public const string TableOfContent = nameof(YamlMime) + ":" + nameof(TableOfContent);
-        public const string XRefMap = nameof(YamlMime) + ":" + nameof(XRefMap);
+        public const string YamlMimePrefix = nameof(YamlMime) + ":";
+        public const string ManagedReference = YamlMimePrefix + nameof(ManagedReference);
+        public const string TableOfContent = YamlMimePrefix + ":" + nameof(TableOfContent);
```
Comment (qinezh) [inline src/Microsoft.DocAsCode.Common/YamlMime.cs]: >":" [](start = 62, length = 3)

remove?

**Comment by superyyrrzz** [comment]
:shipit:

**Comment by hellosnow** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Common/YamlMime.cs [hunk 3]
```diff
@@ -23,7 +24,12 @@ public static string ReadMime(TextReader reader)
             {
                 return null;
             }
-            return line.TrimStart('#').TrimStart(' ');
```
Comment (ansyral) [inline src/Microsoft.DocAsCode.Common/YamlMime.cs]: >TrimStart('#').TrimStart(' ') [](start = 24, length = 29)

how about TrimStart('#', '') ?

Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Common/YamlMime.cs]: I don't want to trim: `# # #`

**Pull request merged**